config/starterkit.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+return [
+
+    /* 
+     * The users' netid may be retrieved from an environment variable
+     * populated by the Apache shibboleth module.  
+     * 
+     * The default env variable is REMOTE USER, but this may be e.g, 
+     * REDIRECT_REMOTE_USER on some servers, depending on how PHP is installed.
+     * 
+     * This configuration allows configuration of the remote_user_variable so it
+     * is not hardcoded in the application. 
+     * 
+     * In project code, use env(config('starterkit.remote_user_variable')) to retrieve the netid
+     *
+     * For local development without shibboleth, you can add
+     * REMOTE_USER=<netid>
+     * to your project .env file to set the user netid. 
+     *
+     */
+    'remote_user_variable' => env("REMOTE_USER_VARIABLE", "REMOTE_USER"),
```

Comment:
I think this PR is a great idea. I just suggest to name it 'remote_user' instead of 'remote_user_variable'. I think 'remote_user_variable' is too long.

Reply:
I worried that it would be too confusing, but as long as people used env(config('starterkit.remote_user')) it would be ok.  Thoughts, @woodseowl  ?

Reply:
I think the long name is appropriate. `remote_user` as a name suggests to me that it is some kind of entity.

Reply:
I think '_variable' does not give any clue that it is local. Maybe  'remote_user_app' then? 
 Another question: do we need a separate "starterkit" config file? Would it be cleaner to use config/app.php. It is almost empty. We are using app.js and app.css for other application customizations.

Reply:
The value is the name of the environment variable used by the server (Apache module / PHP), so it really is a variable.  We could call it remote_user_environment_variable to be most clear.  I think that @woodsowl's idea of putting this in its own package with the middleware is a good one.

Reply:
Since this is in a package we don't have access to config/app.php.  starterkit is the shortname of the package, so that is what the config fille is called.   If we push this into it's own cu-auth package as Eric suggests, then it can go in cu-auth.php (or cuauth.php?  I'm not sure what happens to the dash in the package template).